hbase_sql/src/main/java/com/splicemachine/derby/lifecycle/HEngineSqlEnv.java

```diff
@@ -169,4 +179,177 @@ public ServiceDiscovery serviceDiscovery() {
 
+    /**
+     * Parse a Spark or Hadoop size parameter value, that may use k, m, g or t
+     * to represent kilobytes, megabytes, gigabytes or terabytes, respectively,
+     * and return back the corresponding number of bytes.
+     * @param sizeString the parameter value string to parse
+     * @param defaultValue the default value of the parameter if an invalid
+     *                     <code>sizeString</code> was passed.
+     * @return The value in bytes of <code>sizeString</code>, or <code>defaultValue</code>
+     *         if a <code>sizeString</code> was passed that could not be parsed.
+     */
+    private static long parseSizeString(String sizeString, long defaultValue) {
+        long retVal = defaultValue;
+        Pattern sizePattern = Pattern.compile("([\\d.]+)([kmgt])", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = sizePattern.matcher(sizeString);
+        Map<String, Integer> suffixes = new HashMap<>();
+        suffixes.put("k", 1);
+        suffixes.put("m", 2);
+        suffixes.put("g", 3);
+        suffixes.put("t", 4);
+        if (matcher.find()) {
+            BigInteger value;
+            String digits = matcher.group(1);
+            try {
+              value = new BigInteger(digits);
+            }
+            catch (NumberFormatException e) {
+              return defaultValue;
+            }
+            int power = suffixes.get(matcher.group(2).toLowerCase());
+            BigInteger multiplicand = BigInteger.valueOf(1024).pow(power);
+            value = value.multiply(multiplicand);
+            if (value.compareTo(BigInteger.valueOf(Long.MAX_VALUE)) > 0)
+              return Long.MAX_VALUE;
+            if (value.compareTo(BigInteger.valueOf(1)) < 0)
+              return defaultValue;
+
+            retVal = value.longValue();
+        }
+        else {
+            try {
+                retVal = Long.parseLong(sizeString);
+            }
+            catch (NumberFormatException e) {
+                return defaultValue;
+            }
+            if (retVal < 1)
+                retVal = defaultValue;
+        }
+        return retVal;
+    }
+
+    @Override
+    public int getMaxExecutorCores() {
+        return MAX_EXECUTOR_CORES;
+    }
+
+    /**
+     * Estimate the maximum number of Spark executor cores that could be simultaneously
+     * be running given the current YARN and splice.spark settings.
+     *
+     * @return The maximum number of Spark executor cores.
+     */
+    private static int calculateMaxExecutorCores() {
+        String memorySize = HConfiguration.unwrapDelegate().get("yarn.nodemanager.resource.memory-mb");
+        String sparkDynamicAllocationString = getProperty("splice.spark.dynamicAllocation.enabled");
+        String executorInstancesString = getProperty("splice.spark.executor.instances");
+        String executorCoresString = getProperty("splice.spark.executor.cores");
+        String sparkExecutorMemory = getProperty("splice.spark.executor.memory");
+
```

Comment:
if converted to
`private static int calculateMaxExecutorCores(String memorySize, String sparkDynamicAllocationString, String executorInstancesString,  String executorCoresString, String sparkExecutorMemory)`

we can also unit test this function whose logic isn't trivial

Reply:
I've added unit tests for this method as well.